include/libdnf5/base/goal.hpp

```diff
@@ -354,6 +354,12 @@ class Goal {
     void add_provide_install(
         const std::string & spec, const libdnf5::GoalJobSettings & settings = libdnf5::GoalJobSettings());
 
+    /// Add transaction replay request to the goal.
+    ///
+    /// @param json_serialized_transaction      A string of JSON serialized transaction
+    /// @param settings                         A structure to override default goal settings.
+    void add_transaction_replay(const std::string & json_serialized_transaction, libdnf5::GoalJobSettings & settings);
```

Comment:
What about to replace `replay` by `from_file`? The current PR allows not only replay the transaction but also add additional requests.

Reply:
It is added from string so perhaps something like `add_transaction_actions_from_string()`?

Although I kind of like the `replay` word because it points to the format of the loaded string.

Reply:
The method that creates the string is called `serialize_to_json`. What about to add to the name both worlds. `add_serialized_transaction_from_json`  or shorter `add_serialized_transaction` with skipping type of information.
Anyway `add_transaction_actions_from_string()` is also ok.

Reply:
I like the shorter `add_serialized_transaction()` which drops the JSON. I am thinking about dropping it also from the `serialize_to_json()`.
So that we have just:
- `libdnf5::base::transaction::serialize()`
- `libdnf5::transaction::transaction::serialize()`
- `libdnf5::goal::add_serialized_transaction()`